astropy/coordinates/stokes_coord.py

```diff
@@ -0,0 +1,245 @@
+    # (rather than as name.value)
+    _represent_as_dict_primary_data = "value"
+    # Attributes that should be presented as positional arguments to
+    # the class initializer (which takes "stokes" as an argument, not "value").
+    _construct_from_dict_args = ("value",)
+
+    @property
+    def unit(self):
+        return None
+
+    @property
+    def dtype(self):
+        return self._parent._data.dtype
+
+    def new_like(self, cols, length, metadata_conflicts="warn", name=None):
+        """
+        Return a new StokesCoord instance which is consistent with the
+        input ``cols`` and has ``length`` rows.
+
+        This is intended for creating an empty column object whose elements can
+        be set in-place for table operations like join or vstack.
+
+        Parameters
+        ----------
+        cols : list
+            List of input columns
+        length : int
+            Length of the output column object
+        metadata_conflicts : str ('warn'|'error'|'silent')
+            How to handle metadata conflicts
+        name : str
+            Output column name
+
+        Returns
+        -------
+        col : `~astropy.coordinates.StokesCoord` (or subclass)
+            Empty instance of this class consistent with ``cols``
+
+        """
+        # Get merged info attributes like shape, dtype, format, description, etc.
+        attrs = self.merge_cols_attributes(
+            cols, metadata_conflicts, name, ("meta", "format", "description")
+        )
+
+        # Make an empty StokesCoord.
+        shape = (length,) + attrs.pop("shape")
+        data = np.empty(shape=shape, dtype=dtype)
+        # Get arguments needed to reconstruct class
+        out = self._construct_from_dict(stokes=data)
+
+        # Set remaining info attributes
+        for attr, value in attrs.items():
+            setattr(out.info, attr, value)
+
+        return out
+
+    def get_sortable_arrays(self):
+        """
+        Return a list of arrays which can be lexically sorted to represent
+        the order of the parent column.
+
+        For StokesCoord this is just the underlying values.
+
+        Returns
+        -------
+        arrays : list of ndarray
+        """
+        return [self._parent._data]
+
+
+class StokesCoord(ShapedLikeNDArray):
+    """
+    A representation of stokes coordinates with helpers for converting to profile names.
+
+    Parameters
+    ----------
+    stokes_values : array-like
+        The numeric values representing stokes coordinates.
+    """
+
+    info = StokesCoordInfo()
+
+    def __init__(self, stokes, copy=False):
+        data = np.asanyarray(stokes)
+        if data.dtype.kind == "U":
+            self._data = self._from_symbols(data)
+        else:
+            self._data = data.copy() if copy and data is stokes else data
+
+    @property
+    def shape(self):
+        return self._data.shape
+
+    @property
+    def value(self):
+        return self._data
+
+    @property
```

Comment:
How about just removing this? It is not tested anyway...

Reply:
I shall test it and doc it. I wanted to add it so that in the what's new or whatever I can say "to get the old behaviour back do `stokes.quantity`".

Reply:
Hmm, I would not do that unless people actually shout. After all, it is not much more work to tell them to do `u.Quantity(stokes.value)` or `stokes.value << u.one`. Indeed, they may well mostly do `stk.value` which would still work...

Alternatively, I know it caused problems before, but those may have disappeared with all the fiddling: defining `__array__` to return `self._data`, `u.Quantity(stokes)` should do the right thing too...

Anyway, my sense would be not to add API just because we *think* it may help with backward compatibility...